src/utils/fs-utils.ts

```diff
@@ -0,0 +1,13 @@
+/* eslint-disable import/prefer-default-export */
+import * as fs from "fs/promises"
+
+export async function doesDirectoryExist(path: string): Promise<boolean> {
+  let result = false
+  try {
+    const stat = await fs.stat(path)
+    result = stat.isDirectory()
+  } catch (e) {
+    result = false
+  }
+  return result
+}
```

Comment:
since the caller here only really cares about whether the path, we could wrap this in a `ResultAsync`. the caller only has to call `isOk` and if it's an error, we could easily capture a log here. 

since we are moving to structured logging, this prevents silent failures in our system (`fs.stat` could fail for a wide variety of reasons) and allows us to capture contextual info. wdyt?